internal/handler/favourites.go

```diff
@@ -0,0 +1,296 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"svenvermeulen/platform-go-challenge/internal/auth"
+	"svenvermeulen/platform-go-challenge/internal/repository/audience"
+	"svenvermeulen/platform-go-challenge/internal/repository/chart"
+	"svenvermeulen/platform-go-challenge/internal/repository/favourite"
+	"svenvermeulen/platform-go-challenge/internal/repository/insight"
+	"svenvermeulen/platform-go-challenge/pkg/model"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+type FavouritesHandler struct {
+	favouriteRepository *favourite.Repository
+	audienceRepository  *audience.Repository
+	chartRepository     *chart.Repository
+	insightRepository   *insight.Repository
+}
+
+func NewFavouritesHandler(favouriteRepository *favourite.Repository,
+	audienceRepository *audience.Repository,
+	chartRepository *chart.Repository,
+	insightRepository *insight.Repository) *FavouritesHandler {
+	return &FavouritesHandler{
+		favouriteRepository: favouriteRepository,
+		audienceRepository:  audienceRepository,
+		chartRepository:     chartRepository,
+		insightRepository:   insightRepository,
+	}
+}
+
+// GetUserFavourites godoc
+//
+// @Summary     Get user favourites
+// @Description gets list of favourites for a specified user
+// @Tags        favourites
+// @Produce     json
+// @Success     200	{object}	[]model.UserFavourite
+// @Failure     400
+// @Failure     401
+// @Failure     404
+// @Failure     500
+// @Router      /favourites/:userid [get]
+func (h *FavouritesHandler) GetFavourites(c *gin.Context) {
+	userId, err := auth.GetUserIDFromToken(c)
```

Comment:
I see that c *gin.Context is used here. Could you think of another potential uses of context ?

Reply:
The most obvious example would be binding request/response bodies, accessing parameters and returning HTTP statuses. Other uses are accessing headers, reading client IP.

Reply:
Generally, passing context.Context from handlers to services and repositories is common practice as it helps us with:
- Cancellation Signals, Timeouts and Deadlines, thus managing the request lifecycle
- Passing Request-Scoped Data, like you said
- Concurrency Management
- Graceful Shutdown of the App